ionoscloud/resource_volume.go

```diff
@@ -1005,21 +1005,27 @@ func resolveVolumeImageName(ctx context.Context, client *ionoscloud.APIClient, i
 	}
 
 	if len(*images.Items) > 0 {
-		for _, i := range *images.Items {
+		var partialMatch *ionoscloud.Image
+		for _, image := range *images.Items {
 			imgName := ""
-			if i.Properties != nil && i.Properties.Name != nil && *i.Properties.Name != "" {
-				imgName = *i.Properties.Name
+			image := image
+
+			if image.Properties != nil && image.Properties.Name != nil && *image.Properties.Name != "" {
+				imgName = *image.Properties.Name
 			}
 
-			if imgName != "" && strings.Contains(strings.ToLower(imgName), strings.ToLower(imageName)) && *i.Properties.ImageType == "HDD" && *i.Properties.Location == location {
-				return &i, err
+			if partialMatch == nil && imgName != "" && strings.Contains(strings.ToLower(imgName), strings.ToLower(imageName)) && *image.Properties.ImageType == "HDD" && *image.Properties.Location == location {
```

Comment:
can we add some comments here? The if statements are getting a bit  too complex.